eZ/Publish/Core/MVC/Symfony/SiteAccess/Matcher/HostElement.php

```diff
@@ -37,10 +37,15 @@ class HostElement implements VersatileMatcher
     /**
      * Constructor.
      *
-     * @param int $elementNumber Number of elements to take into account.
+     * @param int|array $elementNumber Number of elements to take into account.
      */
     public function __construct($elementNumber)
     {
+        if (is_array($elementNumber) && isset($elementNumber['value'])) {
```

Comment:
yes, those are the options. And afaik siteaccess matchers are only really *(besides tests)* initiated from config, which is why I prefer option 2. @bdunogier any opinion on this? :)